expression/expression.go

```diff
@@ -1105,6 +1105,8 @@ func scalarExprSupportedByFlash(function *ScalarFunction) bool {
 			tipb.ScalarFuncSig_CastStringAsTime /*, tipb.ScalarFuncSig_CastDurationAsTime, tipb.ScalarFuncSig_CastJsonAsTime*/ :
 			// ban the function of casting year type as time type pushing down to tiflash because of https://github.com/pingcap/tidb/issues/26215
 			return function.GetArgs()[0].GetType().GetType() != mysql.TypeYear
+		case tipb.ScalarFuncSig_CastTimeAsDuration:
+			return sourceType.GetType() == retType.GetType() || retType.GetType() == mysql.TypeDuration
```

Comment:
why `sourceType.GetType() == retType.GetType()`?

Reply:
Sorry
seeing the above code, I thought I had to write it this way
I have modified it